.github/workflows/test.yaml

```
@@ -21,32 +21,56 @@ concurrency:
 
 jobs:
   snap-builds:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     # only build the snap for pull requests, it's not needed on release branches
     # or on master since we have launchpad build recipes which do this already
     if: ${{ github.event_name == 'pull_request' }}
     steps:
     - name: Checkout code
       uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
 
     - name: Build snapd snap
       uses: snapcore/action-build@v1
       with:
         snapcraft-channel: 8.x/stable
 
+    - name: Set test version
+      run: |
+        sudo apt install -y devscripts
+        dch --newversion "1337.$(dpkg-parsechangelog --show-field Version)" "testing build"
+        sg lxd -c 'snapcraft clean snapd'
+
+    - name: Build test snapd snap
+      uses: snapcore/action-build@v1
+      with:
+        snapcraft-channel: 8.x/stable
+
     - name: Check built artifact
       run: |
-        unsquashfs snapd*.snap meta/snap.yaml usr/lib/snapd/
-        if cat squashfs-root/meta/snap.yaml | grep -q "version:.*dirty.*"; then
-          echo "PR produces dirty snapd snap version"
-          cat squashfs-root/usr/lib/snapd/dirty-git-tree-info.txt
-          exit 1
-        elif cat squashfs-root/usr/lib/snapd/info | grep -q "VERSION=.*dirty.*"; then
-          echo "PR produces dirty internal snapd info version"
-          cat squashfs-root/usr/lib/snapd/info
-          cat squashfs-root/usr/lib/snapd/dirty-git-tree-info.txt
-          exit 1
-        fi
+        for snap in snapd*.snap; do
+          case "${snap}" in
+            snapd_1337.*.snap)
+              # Skip test build
+              ;;
+            *)
+              echo "Checking ${snap}"
+              rm -rf squashfs-root
+              unsquashfs "${snap}" meta/snap.yaml usr/lib/snapd/
+              if cat squashfs-root/meta/snap.yaml | grep -q "version:.*dirty.*"; then
```

it's probably ok to just grep both files
```suggestion
              if grep -q "version:.*dirty.*" squashfs-root/meta/snap.yaml squashfs-root/usr/lib/snapd/info; then
```